sbor/src/schema/payload_validation/payload_validator.rs

```diff
@@ -1,12 +1,12 @@
 use crate::rust::prelude::*;
 use crate::traversal::*;
-use crate::typed_traversal::*;
 use crate::*;
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum PayloadValidationError<E: CustomTypeExtension> {
-    TraversalError(TypedTraversalError<E::CustomValueKind>),
+    TraversalError(TypedTraversalError<E>),
     TypeValidationError(TypeValidationError),
+    SchemaInconsistency,
```

Comment:
Yes - this will be used by EG the toolkit.

In these contexts, it might be possible for a schema to be invalid. A toolkit user (such as the Gateway) wouldn't want to have their whole app crash in these cases!

In fact, I'd go as far as saying that the toolkit should never panic - a panic should be reserved for programming errors (like unreachable violations).

Reply:
I'm happy either way. 

We just need to make sure the contract is clear: A) all schema-based validation allows invalid schema; B) payload validation assumes the input schema is valid.

If we go with route A, we need to check if any of our implementation has made assumption about schema validity. 

If we go with route B. we need to make sure all clients do schema check properly upfront (to avoid panics).